oneandone/src/test/java/org/apache/jclouds/oneandone/rest/features/SharedStorageApiLiveTest.java

```diff
@@ -0,0 +1,152 @@
+
+      assertNotNull(imageWithQuery);
+      Assert.assertTrue(imageWithQuery.size() > 0);
+   }
+
+   public void testGet() {
+      SharedStorage result = sharedStorageApi().get(currentSharedStorage.id());
+
+      assertNotNull(result);
+      assertEquals(result.id(), currentSharedStorage.id());
+   }
+
+   @Test(dependsOnMethods = "testGet")
+   public void testUpdate() throws InterruptedException {
+      String updatedName = "Updatedjava";
+
+      SharedStorage updateResult = sharedStorageApi().update(currentSharedStorage.id(), SharedStorage.UpdateSharedStorage.create(updatedName, "desc", null));
+
+      assertNotNull(updateResult);
+      assertEquals(updateResult.name(), updatedName);
+
+   }
+
+   @Test(dependsOnMethods = "testUpdate")
+   public void testAttachServer() throws InterruptedException {
+
+      List<SharedStorage.Server.CreateServer.ServerPayload> servers = new ArrayList<SharedStorage.Server.CreateServer.ServerPayload>();
+      SharedStorage.Server.CreateServer.ServerPayload toAdd = SharedStorage.Server.CreateServer.ServerPayload.create(currentServer.id(), Types.StorageServerRights.R);
+      servers.add(toAdd);
+      SharedStorage updateResult = sharedStorageApi().attachServer(currentSharedStorage.id(), SharedStorage.Server.CreateServer.create(servers));
+
+      assertNotNull(updateResult);
+
+   }
+
+   @Test(dependsOnMethods = "testAttachServer")
+   public void testListServers() {
+      List<SharedStorage.Server> servers = sharedStorageApi().getServers(currentSharedStorage.id());
+
+      assertNotNull(servers);
+      Assert.assertTrue(servers.size() > 0);
+   }
+
+   @Test(dependsOnMethods = "testListServers")
+   public void testServerGet() {
+      SharedStorage.Server result = sharedStorageApi().getServer(currentSharedStorage.id(), currentServer.id());
+
+      assertNotNull(result);
+      assertEquals(result.id(), currentServer.id());
+   }
+
+   @Test(dependsOnMethods = "testServerGet")
+   public void testDetachServer() {
+      SharedStorage result = sharedStorageApi().detachServer(currentSharedStorage.id(), currentServer.id());
+
+      assertNotNull(result);
+      assertEquals(result.id(), currentSharedStorage.id());
+   }
+
+   @Test(dependsOnMethods = "testDetachServer")
+   public void testListAccessCredentials() throws InterruptedException {
+
+      List<SharedStorageAccess> access = sharedStorageApi().getAccessCredentials();
+
+      assertNotNull(access);
+      assertTrue(access.size() > 0);
+   }
+
+   public void testChangePassword() throws InterruptedException {
+      List<SharedStorageAccess> response = sharedStorageApi().changePassword(SharedStorageAccess.UpdateSharedStorageAccess.create("Test123!"));
+
+      assertNotNull(response);
+   }
+
+   @AfterClass(alwaysRun = true)
+   public void teardownTest() throws InterruptedException {
+      sharedStorageApi().delete(currentSharedStorage.id());
+      assertNodeAvailable(currentServer);
```

Comment:
Does it make sense to try to delete the server even if the shared storage delete operation failed? We should try to avoid leaking resources.